pyroma/projectdata.py

```diff
@@ -3,9 +3,69 @@
 import sys
 import logging
 import tokenize
-
+import build
+import email
+import pathlib
+import tempfile
 from copy import copy
 from distutils import core
+from setuptools import config
+
+METADATA_MAP = {
+    "summary": "description",
+    "classifier": "classifiers",
+    "project_url": "project_urls",
+    "home_page": "url",
+}
```

Comment:
Yeah, PEP621 is the aim there. I did check that "summary" wasn't the standard, but I haven't looked any further there yet.

Reply:
It's called `description` in PEP 621, as well as all of the major tools' legacy metadata config, so you're already good there—only core metadata calls it `summary`. Similarly, `classifier` is called `classifiers` in all of them—its just not plural in core metadata because its represented as a multi-use field rather than an array (due to how RFC 2822 works).

What Setuptools calls `project_urls` is called `urls` in PEP 621 and all the other tools, and the legacy `home_page` (what setuptools calls `url`) is not explicitly specified at all in them (though it can be implicitly filled in by tools based on the `urls` value, IIRC).

There are some other things not handled here. What core metadata calls `Description` and setuptools calls `long_description` is called `readme` in PEP 621 and Poetry, and `description-file` in Flit. Also, `Requires-Python` in Core Metadata is `requires-python` in PEP 621 and Flit, but is `python_requires` in setuptools, and dependencies and entry points are rather different as well. See [PEP 621](https://peps.python.org/pep-0621/) for the full details.

Reply:
I will put the PEP621 stuff in a separate PR, maybe next weekend. Partly because full PEP621 will require setuptools > 61, and it feels way to bleeding edge for me, seeing as it was released this Thursday.  :-)

Reply:
@regebro Yeah, I'd wait to adopt PEP 621 for this project's own build config until setuptools support is fully stable, which will probably be at least a few months. The advantage of PEP 517 builds is a compatible version of setuptools is installed automatically for any users installing from source/sdists (probably very few), so you don't need to wait for most systems to be updated to a version that supports it to use such new features.